src/Compiler/Driver/GraphChecking/FileContentMapping.fs

```diff
@@ -302,9 +302,27 @@ let visitSynTypeConstraint (tc: SynTypeConstraint) : FileContentEntry list =
     | SynTypeConstraint.WhereTyparIsEnum(typeArgs = typeArgs) -> List.collect visitSynType typeArgs
     | SynTypeConstraint.WhereTyparIsDelegate(typeArgs = typeArgs) -> List.collect visitSynType typeArgs
 
+/// Special case of `nameof Module` type of expression
+let (|NameofExpr|_|) (e: SynExpr) =
+    let rec stripParen (e: SynExpr) =
+        match e with
+        | SynExpr.Paren(expr = expr) -> stripParen expr
+        | _ -> e
+
+    match e with
+    | SynExpr.App(flag = ExprAtomicFlag.NonAtomic; isInfix = false; funcExpr = SynExpr.Ident nameofIdent; argExpr = moduleNameExpr) ->
+        if nameofIdent.idText <> "nameof" then
+            None
+        else
```

Comment:
Can this be matched inline?

Reply:
I'm not entirely sure what you mean here. I tried something, let me know if that is what you had in mind.

Reply:
I thought the type was a union, so it would be possible to match the field easily. 🙂

Reply:
Yeah. It seems these types were added earlier than struct unions became possible.

Reply:
I've asked @dawedawe to take a look at this in a future PR.